### PR TITLE
Processing `index` and `columns` seperately (and correctly) in `from_tensor`

### LINKED ISSUE #718: [BUG] Creating DataFrame failed if columns provided
**Describe the bug**

Creating DataFrame failed if columns provided.

**To Reproduce**
To help us reproducing this bug, please provide information below:
1. Your Python version
2. The version of Mars you use
3. Versions of crucial packages, such as numpy, scipy and protobuf
4. Full stack of the error.
5. Minimized code to reproduce the error.

```
In [1]: import mars.dataframe as md                                             

In [2]: import mars.tensor as mt                                                

In [3]: df = md.DataFrame(mt.random.rand(10, 3), columns=list('abc'))           
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-3-8015891094de> in <module>
----> 1 df = md.DataFrame(mt.random.rand(10, 3), columns=list('abc'))

~/Workspace/mars/mars/dataframe/initializer.py in __init__(self, data, index, columns, dtype, copy, chunk_size, gpu, sparse)
     29                  chunk_size=None, gpu=None, sparse=None):
     30         if isinstance(data, TENSOR_TYPE):
---> 31             df = from_tensor(data, index=index, columns=columns, gpu=gpu, sparse=sparse)
     32         elif isinstance(data, DATAFRAME_TYPE):
     33             df = data

~/Workspace/mars/mars/dataframe/datasource/from_tensor.py in from_tensor(tensor, index, columns, gpu, sparse)
    135     gpu = tensor.op.gpu if gpu is None else gpu
    136     op = DataFrameFromTensor(dtypes=pd.Series([tensor.dtype] * col_num), gpu=gpu, sparse=sparse)
--> 137     return op(tensor, index, columns)

~/Workspace/mars/mars/dataframe/datasource/from_tensor.py in __call__(self, input_tensor, index, columns)
     51     def __call__(self, input_tensor, index, columns):
     52         if index is not None or columns is not None:
---> 53             if input_tensor.shape != (len(index), len(columns)):
     54                 raise ValueError(
     55                     '({0},{1}) should have the same shape with tensor: {2}'.format(index, columns, input_tensor.shape))

TypeError: object of type 'NoneType' has no len()
```

### PR DESCRIPTION
**Pull request by sighingnow**
## What do these changes do?

When constructing `DataFrame` from `Tensor`, the `index` and `columns` argument should be processed separately. When user feed a custom `index` argument we need to store the underlying value and when the argument is not `pd.Index` we should perform a conversion.

This pr also fixes a bug in `IndexValue.has_value`.

## Related issue number

Fixes #718

### CODE REVIEW

File: mars/dataframe/datasource/from_tensor.py [hunk 1]
```diff
@@ -49,26 +49,32 @@ def _set_inputs(self, inputs):
...
-            # convert to DataFrame
-            index_value = parse_index(pd.RangeIndex(start=0, stop=input_tensor.shape[0]))
-            columns_value = parse_index(pd.RangeIndex(start=0, stop=input_tensor.shape[1]))
 
-        # overwrite index_value and columns_value if user has set them
         if index is not None:
+            if input_tensor.shape[0] != len(index):
+                raise ValueError(
+                    'index {0} should have the same shape with tensor: {1}'.format(index, input_tensor.shape[0]))
+            if not isinstance(index, pd.Index):
```
Comment (qinxuye) [inline mars/dataframe/datasource/from_tensor.py]: I think for now we can convert `index` to pandas Index when it is `pd.Index` and not `mars.core.Base`, we could add support for mars objects later.

File: mars/dataframe/datasource/from_tensor.py [hunk 2]
```diff
@@ -49,26 +49,32 @@ def _set_inputs(self, inputs):
...
+                index = pd.Index(index)
             index_value = parse_index(index, store_data=True)
+        else:
+            index_value = parse_index(pd.RangeIndex(start=0, stop=input_tensor.shape[0]))
+
         if columns is not None:
+            if input_tensor.shape[1] != len(columns):
+                raise ValueError(
+                    'columns {0} should have the same shape with tensor: {1}'.format(columns, input_tensor.shape[1]))
+            if not isinstance(columns, pd.Index):
```
Comment (qinxuye) [inline mars/dataframe/datasource/from_tensor.py]: Ditto.

File: mars/dataframe/datasource/from_tensor.py (hunk 1 above)
Comment (sighingnow) [reply mars/dataframe/datasource/from_tensor.py]: > we can convert `index` to pandas Index when it is `pd.Index` and not `mars.core.Base`

Did you mean "converting mars IndexValue to pandas Index"?

Comment (qinxuye) [reply mars/dataframe/datasource/from_tensor.py]: No, I meant that Mars tileable objects like tensor are not considered here.

Comment (sighingnow) [reply mars/dataframe/datasource/from_tensor.py]: We don't consider mars tileable here. `not instance(index, pd.Index)` means the `index` can be a `list`, or `np.ndarray`.

Comment (qinxuye) [reply mars/dataframe/datasource/from_tensor.py]: Yeah, but tileable objects are not `pd.Index` as well. We could raise `NotImplementedError` here, or they would be converted to `pd.Index`.

Comment (sighingnow) [reply mars/dataframe/datasource/from_tensor.py]: I see, fixed now.

**Pull request merged**